docs/blueprints/EP031.rst

```diff
@@ -249,9 +249,40 @@ V. Events
+| ``deleted``                      | remove flows; delete; archive                                      | remove flows; disable                                 |
++----------------------------------+--------------------------------------------------------------------+-------------------------------------------------------+
+| ``redeployed_link_down``         | redeploy                                                           | same                                                  |
++----------------------------------+--------------------------------------------------------------------+-------------------------------------------------------+
+| ``redeployed_link_up``           | redeploy                                                           | same                                                  |
++----------------------------------+--------------------------------------------------------------------+-------------------------------------------------------+
+| ``error_redeploy_link_down``     | remove flows; deactivate                                           | same                                                  |
++----------------------------------+--------------------------------------------------------------------+-------------------------------------------------------+
+| ``failover_link_down``           | install ingress flows; publish new flows                           | generate subset flows; install flows                  |
++----------------------------------+--------------------------------------------------------------------+-------------------------------------------------------+
+| ``failover_old_path``            | remove old flows; publish old flows (old current_path or failover) | generate subset flows; remove flows                   |
++----------------------------------+--------------------------------------------------------------------+-------------------------------------------------------+
+| ``failover_deployed``            | remove failover; install failover flows; publish old and new flows | generate subset flows; remove old; install new        |
++----------------------------------+--------------------------------------------------------------------+-------------------------------------------------------+
+
+Major challenges to be aware when dealing with **mef_eline** events convergence: a) ensure failover fast convergence, and b) **telemetry_int** isn't differentiating which **mef_eline** path each flow belongs to. Whenever it's desirable that **telemetry_int** only perform a side effect on a subset of the flows, **mef_eline** should facilitate ideally by publishing upfront the set of the flows since **mef_eline** owns the follows and **telemetry_int** essentially follows. In general, the final outcome when handling these events is supposed to only add a few extra milliseconds on top of the existing 2023.2 **mef_eline** flows convergence:
```

Comment:
The simplest implementation would be to always redeploy for any failover event too (remove and reinstall all EVC flows every time), that could work and it'd be much easier to reason and maintain too, but then it would throw away any of the fast convergence part. We can also consider this gradually or not even support fast convergence at all. I was assuming that indeed would be a major requirement. Let me know if that has changed. If it'd be acceptable to always redeploy that's even easier to deliver.  

Would it be acceptable to always redeploy for failover events on `2024.1`?

Reply:
#### Adjacent related notes


When mef_eline publishes the event they'll be queued as expected, and on `telemetry_int` side the idea was to handle them in the event loop with asyncio, so when handling those at least out of order events shouldn't be an issue, and the rest of the concurrent part we're not having many yield points with r+w where a shared state changes. But, certainly it's more much more work and harder to maintain, and, potentially, `kill -9` and/or failed retry DB writes (or any other existing race conditions - looking on mef_eline issues, other than the orphan flows that shouldn't affect anymore I'm not aware of other ones), although they're very rare to happen, but it can leave in a problematic state. Ultimately, consistency check might aid somewhat, but then we also don't have it yet.

Reply:
I see, I forgot that telemetry_int will create more priority and more specific flows for TCP/UDP, which ultimately will make mef_eline flows useless, even after fast failover. Maybe you could also write this on the blueprint, just for the sake of making sure a future reader will keep this in mind. Having said that, I do believe we should keep the failover here. Indeed it is a major requirement.

Reply:
Right, thanks for confirming the requirement @italovalcy. Augmented the higher priority text here on https://github.com/kytos-ng/kytos/pull/468/commits/768a3617cdf6d5234d69dfb165ee4feadd1c053a. Let's keep this way here then for the event involving failover to always publish upfront the related flows just so telemetry_int can react accordingly. In terms of the events, I'm optimistic with this proposted implementation, since the places where mef_eline will publish it'll also send the flows accordingly upfront, so as long as telemetry_int process them in order, we shouldn't expect surprises, other than kill -9 type of stuff where that can also be a problem not only in here but other parts too. I'll make sure to also stress test this part later. 

#### Future side note

Regarding sending flows upfront in the events, other events such as `redeployed_link_up`, `redeployed_link_down`, `deployed`, and other ones, could potentially also publish the flows just so telemetry_int wouldn't need to refetch them, but at the moment, it's not worth it, and let's only keep the failover related ones publishing the flows, just so we minimize refactoring, plus the request that telemetry_int will make and mef_eline DB call on average is expected to be a single or low two digits milliseconds, and these ones here don't have to be as quick as failover, so it's OK. In the future, there's room to potentially optimize here too for telemetry_int (although cutting a few ms here might not even be worth it). But, one step at a time. Let's first confirm in practice the failover ones will succeed and operate quick, correctly and reliably.